frontend/src/components/route/NewRouteMap.js

```diff
@@ -0,0 +1,187 @@
+};
+const options = {
+    styles: MapStyles,
+    disableDefaultUI: true,
+    zoomControl: true,
+}
+
+export default function NewRouteMap({ latitude, longitude }) {
+
+    const history = useHistory();
+    const { addNewRouteAndUpdateAllRoutes } = useContext(RouteContext);
+
+    const [waypoints, setWaypoints] = useState([]);
+    const [routeName, setRouteName] = useState("");
+    const [showName, setShowName] = useState(false);
+
+    const { isLoaded, loadError } = useLoadScript({
+        googleMapsApiKey: process.env.REACT_APP_GOOGLE_MAPS_API_KEY,
+        libraries
+    })
+
+    if (loadError) {
+        return (
+            <p>Error loading map</p>
+        );
+    }
+    if (!isLoaded) {
+        return (
+            <p>Loading map</p>
+        );
+    }
+    if (isLoaded) {
+        return (
+            <MapContainer>
+                {showName ?
+                    <div>
+                        <h1>{routeName}</h1>
+                    </div>
+                    :
+                    <div>
+                        <input
+                        type={"text"} placeholder={"name of your route"}
+                        value={routeName} onChange={event => setRouteName(event.target.value)}
+                        />
+                        <button className={"saveName"} onClick={saveRouteName}>Save route name</button>
+                    </div>
+                }
+                <GoogleMap mapContainerStyle={mapContainerStyle}
+                           zoom={10} center={centerMap()} options={options}
+                           onClick={clickHandler}>
+                    {waypoints.map(waypoint =>
+                        <Marker position={waypoint} key={waypoints.indexOf(waypoint)}
+                                label={labelMarker(waypoints, waypoint)}
+                        />)
+                    }
+                    <Polyline path={waypoints}/>
+                </GoogleMap>
+                <button className={"cancel"} onClick={redirectBackToRoutes}><MdCancel/>Cancel</button>
+                <button className={"done"} onClick={createRoute}><MdDone/>Done</button>
+            </MapContainer>
+        );
+    }
+
+    function centerMap() {
+        if (waypoints.length === 0) {
+            return {
+                lat: latitude,
+                lng: longitude,
+            }
+        }
+        return {
+            lat: waypoints[waypoints.length - 1].lat,
+            lng: waypoints[waypoints.length - 1].lng,
+        }
+    }
+
+    function clickHandler(event) {
+        setWaypoints([...waypoints, {lat: event.latLng.lat(), lng: event.latLng.lng()}]);
+    }
+
+    function labelMarker(collection, item) {
+        if ((collection.indexOf(item)) === 0) {
+            return "Start";
+        }
+        if ((collection.indexOf(item) + 1) === collection.length) {
+            return "End";
+        }
+        return String(collection.indexOf(item));
+    }
+
+    function saveRouteName() {
```

Comment:
Maybe rename to `toggleShowName`? Just from reading the function name I expect it to persistet the route name.